src/operators.jl

```diff
@@ -178,11 +175,7 @@ function Base.:^(lhs::AbstractVariableRef, rhs::Integer)
     elseif rhs == 0
```

Comment:
Now that we're giving up on type stability, should the `rhs==1` case return an affine expression?

Reply:
I talked about this with @ccoffrin. I think we'd like to, but perhaps we should keep that for a separate PR after this is merged, since it's a bit orthogonal, and potentially breaking, depending on how you define things.

Reply:
We would open an issue about that to make sure we don't forget

Reply:
Done https://github.com/jump-dev/JuMP.jl/issues/3454